src/test/java/net/datafaker/providers/base/AviationTest.java

```diff
@@ -18,12 +18,12 @@ void flight_ICAO() {
 
     @Test
     void flight_IATA() {
-        assertThat(aviation.flight("IATA")).matches("[A-Z]{2}[0-9]+");
+        assertThat(aviation.flight("IATA")).matches("[A-Z0-9]{2}[0-9]+");
     }
 
     @Test
     void flight_default() {
-        assertThat(aviation.flight()).matches("[A-Z]{2}[0-9]+");
+        assertThat(aviation.flight()).matches("[A-Z0-9]{2}[0-9]+");
```

Comment:
```suggestion
        assertThat(aviation.flight()).matches("[A-Z0-9]{2}\\d{1,4}");
```
it's better to have regex sync between `shouldNotApplyCachingToMethodsWithParameters` and this test